docs/EVALUATION.md

```diff
@@ -0,0 +1,249 @@
+
+Theoretically it would be possible to partition the evaluation split of one "triggering interval" into multiple evaluation splits for every evaluation point. This would allow us to evaluate the model at every evaluation point with the same samples. The advantage of this is that we could combine these single results to a total performance over the whole triggering interval. The same problems as in [2] would apply though (e.g. weighting of the results). Therefore I suggest treating the two evaluation types separately (also cleaner in code).
+
+> Remark: We would need new evaluation strategies for this as we need intervals with custom bounds on both sides.
+
+```mermaid
+gantt
+    title Pipeline run
+    dateFormat YYYY-MM-DD
+    axisFormat %Y
+    section models
+        Trigger1 :milestone, 1930, 0d
+        model 1          :a1, 1930, 10y
+        Trigger2 :milestone, :after a1
+        model 2          :a2, after a1, 5y
+        Trigger3 :milestone, :after a2
+        model 3          :a3, after a2, 5y
+        Trigger4 :milestone, :after a3
+        model 4          :a4, after a3, 25y
+        Trigger5 :milestone, :after a4
+        model 5          :a5, after a4, 25y
+        Trigger6 :milestone, :after a5
+        model 6          :a6, after a5, 5y
+        Trigger7 :milestone, :after a6
+        model 7          :a7, after a6, 5y
+    section final evaluations
+        Evaluation1 :milestone, 1940, 0d
+        Evaluation2 :milestone, 1945, 0d
+        Evaluation3 :milestone, 1950, 0d
+        Evaluation4 :milestone, 1975, 0d
+        Evaluation5 :milestone, 2000, 0d
+        Evaluation6 :milestone, 2005, 0d
+        Evaluation7 :milestone, 2010, 0d
+    section final evaluation data
+        data1 :e1, 1930, 10y
+        data2 :e2, after e1, 5y
+        data3 :e3, after e2, 5y
+        data4 :e4, after e3, 25y
+        data5 :e5, after e4, 25y
+        data6 :e6, after e5, 5y
+        data7 :e7, after e6, 5y
+    section local evaluations (for model 4)
+        Evaluation4.1 :milestone, 1950, 0d
+        Evaluation4.1 :milestone, 1954, 0d
+        Evaluation4.1 :milestone, 1958, 0d
+        Evaluation4.1 :milestone, 1962, 0d
+        Evaluation4.1 :milestone, 1966, 0d
+        Evaluation4.1 :milestone, 1970, 0d
+    section local evaluation data (for model 4): +- 5 years
+        data4.1 :e4.1, 1947, 6y
+        data4.2 :e4.2, 1951, 6y
+        data4.3 :e4.3, 1955, 6y
+        data4.4 :e4.4, 1959, 6y
+        data4.5 :e4.5, 1963, 6y
+        data4.6 :e4.6, 1967, 6y
+```
+
+### Open Question
+
+1) With the approach of evaluating a model only on future data (where the model is eventually used), it might even be fair not to use a evaluation split as the samples used for evaluation are never part of the training data of a particular model. Agreed?
```

Comment:
If I understand you correctly, this is one particular evaluation strategy you are talking about (evaluate model trained on data up to 2010 on data from 2011 for example). In this case yes, we could use the data we will train the 2011 model on, but I wonder if this will not complicate things, since we do have the nice timeslicing logic of the evaluation dataset. Why not just evaluate(2010 model, (2011, 2012]) or something?

Reply:
Sure, that's possible. For smaller datasets it's beneficial though to have access to as much data as possible when evaluating.

Reply:
Agreed. It depends on the dataset and scenario. I don't think there is one best solution for what evaluation data to use (or even how to sample an evaluation dataset - should it be e.g. balanced across time such that for years where we have less data we also have less evaluation data, or sampled uniformly at random, or with fixed size across years?)